console/src/components/Services/Data/TablePermissions/Actions.js

```diff
@@ -638,7 +638,7 @@ const permChangePermissions = changeType => {
     const limitEnabled = permissionsState.limitEnabled;
 
     const table = permissionsState.table;
-    const role = permissionsState.role;
+    const role = permissionsState.newRole || permissionsState.role;
```

Comment:
This isn't a neat solution. permissionState.role needs to be the source of truth for what role we are setting permissions for. With this solution the display still shows the old role on the UI but applies to the updated value which is confusing.

Better solution is to update value of permissionState.role along when newRole is updated if edit section is open